fabric-renderer-registries-v1/src/main/java/net/fabricmc/fabric/mixin/client/renderer/registry/MixinItemRenderer.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.renderer.registry;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.render.BufferBuilder;
+import net.minecraft.client.render.Tessellator;
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.item.ItemRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.MathHelper;
+
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.CooldownOverlayProperties;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.CountLabelProperties;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.CustomItemOverlayRenderer;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.DurabilityBarProperties;
+import net.fabricmc.fabric.api.client.rendereregistry.v1.item.ItemOverlayRendererRegistry;
+
+@Mixin(ItemRenderer.class)
+public abstract class MixinItemRenderer {
+	@Shadow public float zOffset;
+	@Shadow
+	protected abstract void renderGuiQuad(BufferBuilder buffer, int x, int y, int width, int height, int red, int green, int blue, int alpha);
+
+	@Unique
+	protected void renderGuiQuad(BufferBuilder buffer, int x, int y, int width, int height, int color) {
+		renderGuiQuad(buffer, x, y, width, height, color >> 16 & 255, color >> 8 & 255, color & 255, color >> 24 & 255);
+	}
+
+	/**
+	 * @reason Implement custom item overlay API
+	 * @author ADudeCalledLeo
+	 */
+	@SuppressWarnings("deprecation")
+	@Overwrite
```

Comment:
this piece of mixin seems vulnerable to minecraft updates due to its overwrite nature, though i cannot figure out a way to improve it unfortunately.